lib/get-exporter.js

```diff
@@ -0,0 +1,120 @@
+} = require('fs-extra');
+const { merge, values, isEmpty } = require('omnibelt');
+const allSettledSerial = require('./all-settled-serial-p');
+const { rollbarLog } = require('./rollbar');
+
+const getExporter = ({
+  apiType, commandType, localStatusParams, remoteStatusParams, getData, extraQuery
+}) => {
+  const exporter = async (pattern, command = {}, loadedConfig) => {
+    let didExport = false;
+    const { apiToken, applicationId, api } = loadedConfig ? loadedConfig : await loadConfig();
+    if (!apiToken || !applicationId) { return; }
+    let items;
+    try {
+      let query = { applicationId };
+      if (extraQuery) {
+        query = merge(query, extraQuery);
+      }
+      items = await paginateRequest(api[apiType].get, query);
+    } catch (e) {
+      return logError(e);
+    }
+    const itemsById = mapById(items);
+    const { remoteStatusByFile, localStatusByFile } = await getShallowStatus({
+      items,
+      remoteStatusParams,
+      localStatusParams,
+      dir: commandType
+    });
+    if (command.dryRun) {
+      logResult('DRY RUN');
+    }
+    if (isEmpty(remoteStatusByFile)) {
+      if (!pattern) {
+        return logResult('Missing', `No ${plural(commandType)} found to export.`, 'yellow');
+      } else {
+        return logResult('No Matches', `No ${plural(commandType)} found that match this pattern ${pattern}`, 'yellow');
+      }
+    }
+    const downloadResults = await allSettledSerial(async (remoteStatus) => {
+      logProcessing(remoteStatus.file);
+      if (!command.force) {
+        const conflict = localStatusByFile[localStatusByFile.indexOf(remoteStatus.file)] && remoteStatus.status === 'found';
+        if (conflict) {
+          const { handleConflict } = await inquirer.prompt([{
+            name: 'handleConflict',
+            type: 'confirm',
+            message: `The file ${remoteStatus.file} already exists, do you want to overwrite it?`
+          }]);
+          if (!handleConflict) {
+            return logResult('skipped', remoteStatus.file, 'yellow');
+          }
+        }
+      }
+      if (remoteStatus.status === 'missing' && !command.dryRun) {
+        try {
+          if (await pathExists(remoteStatus.file)) {
+            const { canDelete } = await inquirer.prompt([{
+              name: 'canDelete',
+              type: 'confirm',
+              message: `The file ${remoteStatus.file} is missing remotely, do you want to delete it locally?`
+            }]);
+            if (canDelete) {
+              await remove(remoteStatus.file);
+              return logResult('deleted', remoteStatus.file, 'yellow');
+            } else {
+              return logResult('unmodified', remoteStatus.file);
+            }
+          }
+        } catch (e) {
+          return logError(`An Error occurred when trying to delete file ${remoteStatus.file} with the message ${e.message}`);
+        }
+      }
+      if (!command.dryRun) {
+        await ensureDir(path.dirname(remoteStatus.file));
```

Comment:
I'm not sure you need this, the reason we do it for download is because of files.

Reply:
True. It fails from the missing dir before it reaches this point anyway.

Reply:
Do we care if the user deletes the directory manually? Because without this we just error if, for some reason, they rm'd it. Maybe they don't deserve to be treated nicely in that case.

Reply:
Well if the user did not reconfigure their project then they don't have the directory, I would move this to the top of the function. If they delete this directory while exporting the data tables that's their own fault.